### PR TITLE
Clear only the current shard's schemacopy rows  during schema reload

### LINKED ISSUE #8510: schema tracking not working in rails tests
#### Overview of the Issue

Rails tests fail in the sharded environment with Gen4Fallback and schema-tracking because schema tracking isn't working properly.

1. A new table is created from the tests named `employee1s`
2. Schema tracker picks up this change and adds it to the vschema table in the vttablet. The following rows are read from the `_vt.schemacopy` table in the vttablet
```
[VARCHAR("employee1s") VARCHAR("id") VARCHAR("bigint")] 
[VARCHAR("employee1s") VARCHAR("company1_id") VARCHAR("bigint")] 
[VARCHAR("employee1s") VARCHAR("created_at") VARCHAR("datetime")] 
[VARCHAR("employee1s") VARCHAR("updated_at") VARCHAR("datetime")]
```
3. vtgate picks up that these tables are updated
```
E0721 08:23:46.057639     757 tracker.go:159] updated tables in schema[account2s account3s account4s account5s account_histories active_storage_attachments active_storage_blobs appointments ar_internal_metadata assemblies assemblies_parts assembly2s assembly2s_part2s assembly3s assembly4s assembly4s_part4s assembly5s assembly5s_part5s author2s author3s author4s author5s authors book2s book3s book4s book5s book6s book6s_order2s books chapters company1s computers customer2s customers employee1s employee2s employee3s employees invoices library manifests manufacturers manufacturers_part5s markets microposts order2s part2s part3s part4s part5s parts patients physicians pictures products puppers puppers_friends relationships reviews schema_migrations supplier2s supplier3s supplier4s supplier5s suppliers trackpads transactions user111s user112s users]
```
4. but on querying the `_vt.schemacopy` table for change updates, there are none
```
E0721 08:23:46.061675     757 tracker.go:182] tables found in schemacopy&{[name:"table_name" type:VARCHAR name:"column_name" type:VARCHAR name:"data_type" type:VARCHAR] 0 0 []  2}
```
5. query which requires schema tracking to expand `*` fails in Gen4 - 
```
E0721 08:23:44.296253     757 fallback_planner.go:54] Gen4 failed planning - Code: UNIMPLEMENTED
gen4 does not yet support: *sqlparser.StarExpr in select list
	vitess.io/vitess/go/vt/vtgate/planbuilder/abstract/queryprojection.go:69 +0xb31
E0721 08:23:44.296422     757 executor.go:210] Executed query :-SELECT `employee1s`.* FROM `employee1s` ORDER BY `employee1s`.`id` DESC LIMIT 1Error -unsupported: in scatter query: order by must reference a column in the select list: employee1s.id desc
```

#### Reproduction Steps

Run rails tests from rails-sharded-gen4 branch of vitess-framework-testing repo. https://github.com/planetscale/vitess-framework-testing/tree/rails-sharded-gen4

#### Binary version
Run against the latest vttestserver image built from main, possibly with additional log messages

### PR DESCRIPTION
**Pull request by frouioui**
## Description

This pull request makes VTTablet clear only the rows that match the current database name and the updated tables in `schemacopy` ; instead of clearing all the rows that match the updated tables. This will be useful for @GuptaManan100 when testing the Rails guide against a sharded cluster. 

VTTestServer uses a single MySQL instance where the different shards are suffixed in the `table_schema` column by their name (`-80`, `80-`, etc). Removing all the rows from the `schemacopy` tables was making the shards not agreeing and made them repetitively send schema updates to VTGate.

After testing VTTestServer with schema tracking enabled: it is preferable to lower the schema update interval in the VTTablets or add multiple timeouts in the Rails test suite. 

## Related Issue(s)

Resolves #8510

## Checklist
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

**Review by GuptaManan100 — APPROVED** [review]
LGTM!

File: go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go [hunk 1]
```diff
@@ -165,6 +165,37 @@ func TestMain(m *testing.M) {
...
+
+	time.Sleep(2 * time.Second)
+
+	assertMatches(t, conn, "select * from test_table", `[]`)
+	assertMatches(t, connShard1, "select * from test_table", `[]`)
+	assertMatches(t, connShard2, "select * from test_table", `[]`)
+
+	exec(t, conn, "drop table test_table")
+
+	time.Sleep(2 * time.Second)
```
Comment (systay) [inline go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go:196]: why this extra sleep?

Comment (frouioui) [reply go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go:196]: Schema tracking needs enough time to propagate the schema changes implied by `DROP TABLE`

Comment (frouioui) [reply go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go:196]: Reload interval is set to `0.1` second, `2` seconds is plenty and should not flake the test

Comment (harshit-gangal) [reply go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go:196]: we have assertmatcheswithtimeout for that.

**Review by harshit-gangal — APPROVED** [review]

File: go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go (hunk 1 above)
Comment (frouioui) [reply go/test/endtoend/vtgate/schematracker/sharded/st_sharded_test.go:196]: The function `assertMatchesWithTimeout` delays the call to `exec` while making sure the call does not exceed a certain timeout, the goal of the `time.Sleep` here is a little different as I am trying to wait after the completion of the `exec`

**Pull request merged**